enclave-runtime/src/initialization/mod.rs

```diff
@@ -175,14 +176,24 @@ pub(crate) fn init_enclave(
 	GLOBAL_TOP_POOL_AUTHOR_COMPONENT.initialize(top_pool_author.clone());
 
 	let getter_executor = Arc::new(EnclaveGetterExecutor::new(state_observer));
-	let io_handler =
-		public_api_rpc_handler(top_pool_author, getter_executor, shielding_key_repository);
+
+	let mut io_handler = IoHandler::new();
+	add_common_api(&mut io_handler, top_pool_author, getter_executor, shielding_key_repository);
+
+	#[cfg(feature = "sidechain")]
+	{
+		use crate::initialization::global_components::EnclaveSidechainBlockImportQueue;
+		use its_rpc_handler::add_sidechain_api;
+		let sidechain_block_import_queue = Arc::new(EnclaveSidechainBlockImportQueue::default());
+		GLOBAL_SIDECHAIN_IMPORT_QUEUE_COMPONENT.initialize(sidechain_block_import_queue);
+		let sidechain_import_queue = GLOBAL_SIDECHAIN_IMPORT_QUEUE_COMPONENT.get()?;
+		let top_pool_author = GLOBAL_TOP_POOL_AUTHOR_COMPONENT.get()?;
+		add_sidechain_api(&mut io_handler, top_pool_author, sidechain_import_queue);
```

Comment:
There is one failing integration test for offchain worker.
If i'm not mistaken this sidechain api is used by offchain worker.
It looks like it cannot be feature guarded like this :)

Reply:
direct top pool api added for non-sidechain nodes too